build: GOOS=js go build ./... fails — REPL drags in chzyer/readline (no wasm support)

### What

`GOOS=js GOARCH=wasm go build ./...` fails to compile — the root `main` package pulls in the REPL, which imports `github.com/chzyer/readline`, and readline has no `js`/`wasm` support.

```
$ GOOS=js GOARCH=wasm go build ./...
# github.com/chzyer/readline
.../readline/utils.go:81:29: undefined: State
.../readline/operation.go:234:4: undefined: ClearScreen
.../readline/readline.go:129:20: undefined: GetScreenWidth
.../readline/readline.go:132:22: undefined: DefaultIsTerminal
.../readline/terminal.go:51:2: undefined: SuspendMe
... (12 errors)
```

### Root cause

`lg_repl.go` is tagged `//go:build !plan9`, so it compiles under `GOOS=js`. It imports `chzyer/readline`, which only builds for native terminals. `lg.go` (the root `main`) has no build constraint, so a wasm build of the root package drags the REPL — and readline — in.

### Impact

Build/CI hygiene, not a runtime bug. The shipped `lg -w` bundle is unaffected: the bundler compiles a generated `main` (the `wasmMainTmpl` temp tree) plus `pkg/rt`, not the root package or the REPL. So `pkg/rt` and the bundle build clean for wasm; only `go build ./...` under `GOOS=js` breaks. It surfaces as noise when cross-checking wasm builds locally.

### Suggested fix

Mirror the existing plan9 split. `repl(ctx *compiler.Context)` already has a plan9 stub in `lg_repl_plan9.go`; the readline-backed implementation in `lg_repl.go` can take the same treatment:

- tag `lg_repl.go` `//go:build !plan9 && !js`, and
- provide a `js`-tagged stub `repl()` (or fold js into the existing stub's constraint).

That keeps the REPL and readline out of any `GOOS=js` build while leaving native untouched.

Will send a PR. Let me know if anything looks off with the approach.

--- a/lg_repl.go
+++ b/lg_repl.go
@@ -1,4 +1,4 @@
-//go:build !plan9
+//go:build !plan9 && !js
 
 /*
  * Copyright (c) 2021 Marcin Gasperowicz <xnooga@gmail.com>

--- a/lg_repl_stub.go
+++ b/lg_repl_stub.go
@@ -1,4 +1,4 @@
-//go:build plan9
+//go:build plan9 || js
 
 /*
  * Copyright (c) 2021 Marcin Gasperowicz <xnooga@gmail.com>
@@ -18,10 +18,11 @@ import (
 	"github.com/nooga/let-go/pkg/vm"
 )
 
-// repl is a minimal line-by-line REPL for plan9. The alimpfard/line readline
-// library depends on termios/ioctl (not available on plan9), so this build
-// reads from stdin via bufio.Scanner — no completion, no syntax highlighting,
-// no in-line editing. Use rio's edit-line history instead.
+// repl is a minimal line-by-line REPL for platforms without readline. The
+// chzyer/readline library depends on termios/ioctl, unavailable on plan9 and
+// js/wasm, so these builds read from stdin via bufio.Scanner — no completion,
+// no syntax highlighting, no in-line editing. (On js the root binary isn't the
+// REPL entry at all; this just keeps GOOS=js go build ./... compiling.)
 func repl(ctx *compiler.Context) {
 	scanner := bufio.NewScanner(os.Stdin)
 	scanner.Buffer(make([]byte, 64*1024), 1024*1024)

